### PR TITLE
refactor: use std::make_unique<T>

### PR DESCRIPTION
**Pull request by miniak**
- replace `std::unique_ptr<T> xyz(new T(...))` with `auto xyz = std::make_unique<T>(...)` for shorter and safer code
- don't allocate on the heap when stack allocation is more suitable

### CODE REVIEW

File: atom/renderer/renderer_client_base.cc [hunk 1]
```diff
@@ -136,9 +136,7 @@ void RendererClientBase::RenderThreadStarted() {
 #if defined(OS_MACOSX)
   base::CommandLine* command_line = base::CommandLine::ForCurrentProcess();
   bool scroll_bounce = command_line->HasSwitch(switches::kScrollBounce);
-  base::ScopedCFTypeRef<CFStringRef> rubber_banding_key(
-      base::SysUTF8ToCFStringRef("NSScrollViewRubberbanding"));
-  CFPreferencesSetAppValue(rubber_banding_key,
+  CFPreferencesSetAppValue(CFSTR("NSScrollViewRubberbanding"),
```
Comment (miniak) [inline atom/renderer/renderer_client_base.cc]: @zcbenz are you ok with this change?

File: atom/browser/ui/file_dialog_mac.mm [hunk 2]
```diff
@@ -85,13 +85,9 @@ void SetAllowedFileTypes(NSSavePanel* dialog, const Filters& filters) {
   // Create array to keep file types and their name.
   for (const Filter& filter : filters) {
     NSMutableSet* file_type_set = [NSMutableSet set];
-    base::ScopedCFTypeRef<CFStringRef> name_cf(
-        base::SysUTF8ToCFStringRef(filter.first));
-    [filter_names addObject:base::mac::CFToNSCast(name_cf.get())];
+    [filter_names addObject:@(filter.first.c_str())];
```
Comment (miniak) [inline atom/browser/ui/file_dialog_mac.mm]: @zcbenz are you ok with this change?

**Comment by miniak** [comment]
@ckerr this one was also funny
```
std::unique_ptr<base::DictionaryValue> empty_info(new base::DictionaryValue);
Browser::Get()->DidFinishLaunching(*empty_info);
```

File: atom/browser/atom_browser_main_parts.cc [hunk 3]
```diff
@@ -223,8 +223,7 @@ void AtomBrowserMainParts::PreMainMessageLoopRun() {
 #if !defined(OS_MACOSX)
   // The corresponding call in macOS is in AtomApplicationDelegate.
   Browser::Get()->WillFinishLaunching();
-  std::unique_ptr<base::DictionaryValue> empty_info(new base::DictionaryValue);
```
Comment (miniak) [inline atom/browser/atom_browser_main_parts.cc]: @ckerr this one was also funny

Comment (deepak1556) [reply atom/browser/atom_browser_main_parts.cc]: there is also another instance at https://github.com/electron/electron/blob/master/atom/browser/mac/atom_application_delegate.mm#L65

**Comment by jkleinsc** [comment]
@miniak can you add a short description to this PR?  Those short descriptions are really helpful when we put together release notes.  Thanks!

**Comment by miniak** [comment]
@jkleinsc done

File: atom/browser/atom_browser_main_parts.cc (hunk 3 above)
Comment (miniak) [reply atom/browser/atom_browser_main_parts.cc]: done

File: atom/renderer/renderer_client_base.cc (hunk 1 above)
Comment (zcbenz) [reply atom/renderer/renderer_client_base.cc]: I'm not familiar with this, but from docs of [CFSTR](https://developer.apple.com/documentation/corefoundation/cfstr?language=objc) this looks fine.

**Pull request merged**